dev/com.ibm.ws.org.apache.cxf.cxf.rt.rs.client.3.2/src/org/apache/cxf/jaxrs/client/CompletionStageRxInvokerImpl.java

```diff
@@ -0,0 +1,215 @@
+        return method(name, Response.class);
+    }
+
+    @Override
+    public CompletionStage<Response> method(String name, Entity<?> entity) {
+        return method(name, entity, Response.class);
+    }
+
+    @Override
+    public <T> CompletionStage<T> method(String name, Entity<?> entity, Class<T> responseType) {
+      //Liberty code change start
+        if (isExecutorNull) {
+         // Run on Liberty executor thread and do not propagate the thread context
+            return ((ManagedExecutor) defaultManagedExecutor).supplyAsync(() -> wc.sync().method(name, entity, responseType));            
+        }      
+        if (isManagedExecutor) {
+            return ((ManagedExecutor) ex).supplyAsync(() -> wc.sync().method(name, entity, responseType));
+        }
+      //Liberty code change end
+        return CompletableFuture.supplyAsync(() -> wc.sync().method(name, entity, responseType), ex);
+    }
+
+    @Override
+    public <T> CompletionStage<T> method(String name, Entity<?> entity, GenericType<T> responseType) {
+      //Liberty code change start
+        if (isExecutorNull) {
+         // Run on Liberty executor thread and do not propagate the thread context
+            return ((ManagedExecutor) defaultManagedExecutor).supplyAsync(() -> wc.sync().method(name, entity, responseType));
+        }      
+        if (isManagedExecutor) {
+            return ((ManagedExecutor) ex).supplyAsync(() -> wc.sync().method(name, entity, responseType));            
+        }
+      //Liberty code change end
+        return CompletableFuture.supplyAsync(() -> wc.sync().method(name, entity, responseType), ex);
+    }
+
+    @Override
+    public <T> CompletionStage<T> method(String name, Class<T> responseType) {
+      //Liberty code change start
+        if (isExecutorNull) {
+         // Run on Liberty executor thread and do not propagate the thread context
+            return ((ManagedExecutor) defaultManagedExecutor).supplyAsync(() -> wc.sync().method(name, responseType));
+        }      
+        if (isManagedExecutor) {            
+            return ((ManagedExecutor) ex).supplyAsync(() -> wc.sync().method(name, responseType));
+        }
+      //Liberty code change end
+        return CompletableFuture.supplyAsync(() -> wc.sync().method(name, responseType), ex);
+    }
+
+    @Override
+    public <T> CompletionStage<T> method(String name, GenericType<T> responseType) {
+      //Liberty code change start
+        if (isExecutorNull) {
+         // Run on Liberty executor thread and do not propagate the thread context
+            return ((ManagedExecutor) defaultManagedExecutor).supplyAsync(() -> wc.sync().method(name, responseType));
+        }      
+        if (isManagedExecutor) {
+            return ((ManagedExecutor) ex).supplyAsync(() -> wc.sync().method(name, responseType));
+        }
+      //Liberty code change end
+        return CompletableFuture.supplyAsync(() -> wc.sync().method(name, responseType), ex);
+    }
```

Comment:
It looks like all of the `method` methods are essentially doing the same thing.  You could probably save some code by adding something like this:
```
private <T> CompletionStage<T> supplyAsync(Supplier<T> supplier) {
        if (isExecutorNull) {
          // Run on Liberty executor thread and do not propagate the thread context
             return ((ManagedExecutor) defaultManagedExecutor).supplyAsync(supplier);
         }      
         if (isManagedExecutor) {
             return ((ManagedExecutor) ex).supplyAsync(supplier);
         }
         return CompletableFuture.supplyAsync(supplier, ex);
}
```

then all of the `method` methods could just call:
`return supplyAsync( () -> wc.sync().method(name, responseType) );` (with the appropriate parameters for each method.